### DISCUSSION TITLE
AlmaLinux9/Grub2/PXE server iPXE Intel i350 failure

### DISCUSSION THREAD

**Discussion by rweisman** [ipxe/ipxe#854] [Q&A]
Ok this is a strange one.
I uEFI pxe boot and get a menu. When I attempt to load iPXE it fails, as it does not receive an IP.

My simple test is a file that just starts iPXE in shell mode.
When I enter dhcp from the prompt, it times out. I do see a DHCPOFFER and ACK in the var/log/messages file.

I am using dnsmasq as the DHCP server.

My previous PXE server was running Centos8 and had no issues.  

My .efi file was built with snponly.efi. I also used ipxe.efi - Same Results
The i350 is a 4 port adding card, in a Dell R740xd system.
When I used a 10Gb dac cable on the base board 25Gb Broadcom it also works.

Attached is the ifstat output
![ifstat](https://user-images.githubusercontent.com/116122090/213467106-b477f893-97ec-4683-94ee-6ca35d591f04.JPG)

**Comment by mcb30** [comment]
What are you using to provide the menu from which you load iPXE?

Your screenshot seems to show successfully transmitted and received packets. Could you please try building with `DEBUG=netdevice:3,udp,dhcp` to get more information?

**Comment by rweisman** [comment]
Michael,
Thank you for the reply.
I am using Grub2 (bootx64.efi) under AlmaLinux.
Here is the Grub menu item
menuentry 'Debug iPXE' {
        chainloader /debugV2.efi
}
After I enter the dhcp command the debug is causing a lot of messages and
it is constantly scrolling.  Not sure how to get that information to you.

Here is a screen cap I grabbed while it was flying by.
[image: image.png]

It is strange as this works with an older version of Linux (Centos8).


In our production environment we have a riety of different manufacturers
and network cards.   This is the only one (So Far) that I am having this
issue with.  It also works with the 25G Baseboard Broadcom NiC.  So far
only an iPXE problem which I use  for sanboot and wimboot.

Again,

Thanks




On Thu, Jan 19, 2023 at 7:42 AM Michael Brown ***@***.***>
wrote:

> What are you using to provide the menu from which you load iPXE?
>
> Your screenshot seems to show successfully transmitted and received
> packets. Could you please try building with DEBUG=netdevice:3,udp,dhcp to
> get more information?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ipxe/ipxe/discussions/854#discussioncomment-4728403>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A3V6D2UZIVF4GDJXQFS6YZTWTFHBLANCNFSM6AAAAAAUAMWMZU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Randy Weisman
***@***.***

**Reply by stappersg** [reply]
> Not sure how to get that information to you. Here is a screen cap I grabbed while it was flying by. [image: image.png]

Please visit this akward website and "post" the image here.   ( In other words:   Do known that it only looks if github.com support email updates  (read: github.com does not care about email updates))

**Reply by mcb30** [reply]
> Michael, Thank you for the reply. I am using Grub2 (bootx64.efi) under AlmaLinux.

OK.  This is the first place to start.  Please remove Grub2 from the test, and have your UEFI BIOS load iPXE directly.  This will simplify the test setup.

**Reply by rweisman** [reply]
As I expected, booting without grub2 worked.   That unfortunately is not a workaround for my environment.  I also find it odd that ipxe and grub2 work together with a different NIC.

Thanks

**Reply by mcb30** [reply]
You mentioned that your previous PXE server (running CentOS 8) worked with no problems, presumably with the same iPXE binary.

This, combined with the fact that it works when GRUB 2 is removed from the setup, very strongly suggests that the problem is caused by a change in the version of GRUB 2 that you are using.  You could try bisecting the GRUB 2 version to find the problematic commit.

**Reply by rweisman** [reply]
I can't argue with that, but the majority of our units (Various vendors) work fine with the Grub2 / ipxe   Even this unit which has a Broadcom 25Gb baseboard  adapter works when connected to that.  Just the Intel Nic in this configuration is the issue.  That is why I am so confused.

**Reply by mcb30** [reply]
Understood, but at this point there's nothing to suggest it's an iPXE problem.  If I were debugging this myself, I'd have to start looking at GRUB 2 first.

**Reply by NiKiZe** [reply]
So snponly.efi fails, which might be because grub leaves the NIC in unexpected state on this specific firmware. Try bios update? Also have you tried to boot ipxe.efi from grub instead, which could use iPXE native drivers instead of SNP/NII driver.

**Comment by rweisman** [comment]
Sorry,
Attached is a screen capture of debug messages as they fly by.
![debugcap](https://user-images.githubusercontent.com/116122090/213559090-9b86b1d3-15e7-4932-b032-a6099e8df2b2.JPG)

**Reply by rweisman** [reply]
Sorry.. That was a cap from a working VM.  Here's the Dell with i350 card caps.
![debugcap](https://user-images.githubusercontent.com/116122090/213567520-48d8c417-96ca-4b56-8e11-2a28f81aa7f7.JPG)
![debugcap1](https://user-images.githubusercontent.com/116122090/213570339-70f0e6f4-fea4-453c-af62-1d733ed8e306.JPG)

**Reply by stappersg** [reply]
Here your human  OCR, Optical Character Recognisance.
The https://ipxe.org/380f6093 is also in the original posting.
